cmake/configure/TestBoostBug/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+CMAKE_MINIMUM_REQUIRED(VERSION 2.8.8)
+PROJECT(TestBoost)
+
+# use, i.e. don't skip the full RPATH for the build tree
+SET(CMAKE_SKIP_BUILD_RPATH  FALSE)
+
+# when building, don't use the install RPATH already
+# (but later on when installing)
+SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+
+SET(CMAKE_MACOSX_RPATH TRUE)
+
+SET(BOOST_INCLUDE_DIRS CACHE STRING "The include paths for BOOST")
+SET(BOOST_LIBRARIES CACHE STRING "The BOOST libraries to use")
+
+# We assume the BOOST_LIBRARIES to be separated by '|'
+# This matches the change in the separator we do in
+# ../configure_boost.cmake
+STRING(REPLACE "|" ";" BOOST_LIBRARIES_SEPARATED ${BOOST_LIBRARIES})
```

Comment:
p.s. I checked that boost libs here look complete:

```
/Users/davydden/spack/opt/spack/darwin-highsierra-x86_64/clang-9.0.0-apple/boost-1.65.1-yn2kzyc572dum542fgbz5ux4gqbs6ddc/lib/libboost_iostreams-mt.dylib;/Users/davydden/spack/opt/spack/darwin-highsierra-x86_64/clang-9.0.0-apple/boost-1.65.1-yn2kzyc572dum542fgbz5ux4gqbs6ddc/lib/libboost_serialization-mt.dylib;/Users/davydden/spack/opt/spack/darwin-highsierra-x86_64/clang-9.0.0-apple/boost-1.65.1-yn2kzyc572dum542fgbz5ux4gqbs6ddc/lib/libboost_system-mt.dylib;/Users/davydden/spack/opt/spack/darwin-highsierra-x86_64/clang-9.0.0-apple/boost-1.65.1-yn2kzyc572dum542fgbz5ux4gqbs6ddc/lib/libboost_thread-mt.dylib;/Users/davydden/spack/opt/spack/darwin-highsierra-x86_64/clang-9.0.0-apple/boost-1.65.1-yn2kzyc572dum542fgbz5ux4gqbs6ddc/lib/libboost_regex-mt.dylib;/Users/davydden/spack/opt/spack/darwin-highsierra-x86_64/clang-9.0.0-apple/boost-1.65.1-yn2kzyc572dum542fgbz5ux4gqbs6ddc/lib/libboost_chrono-mt.dylib;/Users/davydden/spack/opt/spack/darwin-highsierra-x86_64/clang-9.0.0-apple/boost-1.65.1-yn2kzyc572dum542fgbz5ux4gqbs6ddc/lib/libboost_date_time-mt.dylib;/Users/davydden/spack/opt/spack/darwin-highsierra-x86_64/clang-9.0.0-apple/boost-1.65.1-yn2kzyc572dum542fgbz5ux4gqbs6ddc/lib/libboost_atomic-mt.dylib
```

Ran out of ideas, just disable the test on macOS....

Reply:
I am pretty sure that I just don't know how to set the RPATH for macOS correctly. Have you ever created a shared library using cmake on macOS and linked an executable to it successfully?

Reply:
> Have you ever created a shared library using cmake on macOS and linked an executable to it successfully?

sure, i do it all the time with my own code, which is a library + executable and other deal.II based projects of the same kind. It's always
```
ADD_LIBRARY(${TARGETLIB} SHARED
    ${TARGET_SRC}
  )

ADD_EXECUTABLE(${TARGET}
    source/main.cc
  )

DEAL_II_SETUP_TARGET(${TARGETLIB})
DEAL_II_SETUP_TARGET(${TARGET})

TARGET_LINK_LIBRARIES(${TARGET}
    ${TARGETLIB}
  )
```

But of course the same should be done outside of deal.II's macros. 

That's why I said that ideally you would not pass libraries, but actually use already found library to use it in `target_link_libraries`.